src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java

```diff
@@ -73,6 +102,26 @@ public CommandManagerWindow(CommandManager commandManager) {
         content.add(btnClearObservers, c);
     }
 
+    private void loadCommandFromFile() {
+        if (commandFilePath.getText().isEmpty()) {
+            return;
+        }
+
+        if (!commandFilePath.getText().endsWith(".json")) {
+            logger.warning("Unsupported file format");
+            return;
+        }
+
+        CommandLoader loader = new JsonCommandLoader();
```

Comment:
supported formats could be handled externally (via loaders "factory"). It would provide OCP and follow SRP.